source/expressions.tex

```diff
@@ -7310,6 +7310,10 @@
 In simple assignment (\tcode{=}), the object referred to by the left operand
 is modified\iref{defns.access}
 by replacing its value with the result of the right operand.
+\begin{note}
+The object is modified even if the value of the right operand
+compares equal\iref{expr.eq} to the object's old value.
```

Comment:
I don't think this redundancy makes "the object...is modified" any clearer. If anything, it makes me think I must have missed some subtlety in the normative wording.

Reply:
How would you phrase it then? The point this is meant to get across is that even if `x` is `0`, `x = 0` is considered to be modification. This is not obvious from the term "modify" itself, since it is defined as "make partial or minor changes to (something)" in English.

Reply:
"modify" comes with a cross-reference after it, so readers are aware this is a technical term that has more than the English meaning.  If you want to add a cross-reference to the data races section, I'm fine with it.  However, there is no hint in the sentence that the "modify" would be somehow conditional on the values involved, and the "replace" sentence is pretty straightforward, so I'm not seeing a need for a note.

Again, if you wish to suggest adding more cross-reference to add more load to the word "modify", go for it.

Reply:
Ok, so admittedly, [expr.ass] was probably not the right place to approach it. The PR now tackles the problem in [intro.races].